### PR TITLE
Fix softmax and add test

### PR DESCRIPTION
**Pull request by mvafin**

### CODE REVIEW

File: src/frontends/pytorch/src/op_table.cpp [hunk 1]
```diff
@@ -309,7 +309,7 @@ const std::map<std::string, CreatorFunction> get_supported_ops() {
          [](NodeContext& context) -> OutputVector {
              auto axis = context.const_input<int64_t>(1);
              return {
-                 context.mark_node(std::make_shared<opset8::Softmax>(context.get_input(0), static_cast<size_t>(axis)))};
+                 context.mark_node(std::make_shared<opset8::Softmax>(context.get_input(0), axis))};
```
Comment (slyalin) [inline src/frontends/pytorch/src/op_table.cpp:312]: Does it really fix something?

**Review by slyalin — APPROVED** [review]

File: src/frontends/pytorch/src/op_table.cpp (hunk 1 above)
Comment (mvafin) [reply src/frontends/pytorch/src/op_table.cpp:312]: It should fix negative axis case. Because `opset8::Softmax` support negative axis.
